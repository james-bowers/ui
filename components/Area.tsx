import * as React from "react";
import styled, { css } from "./styled";

const sharedAreaStyle = css`
  width: 100%;
`;

const HeroArea = styled.div`
  background: ${props => props.theme.primaryColor};
  color: ${props => props.theme.primaryColorInverted};
  ${sharedAreaStyle}
`;

const StandardArea = styled.div`
  ${sharedAreaStyle}
`;

let StyledPadding = styled.div`
  &.padding_1 {
    padding: ${props => props.theme.spacing * 5}px;
  }

  &.padding_2 {
    padding: ${props => props.theme.spacing * 10}px;
  }

  &.padding_3 {
    padding: ${props => props.theme.spacing * 15}px;
  }

  &.padding_4 {
    padding: ${props => props.theme.spacing * 20}px;
  }

  &.padding_5 {
    padding: ${props => props.theme.spacing * 25}px;
  }
`;

export function Padding(props: {
  children: any;
  multiplier: 1 | 2 | 3 | 4 | 5;
}) {
  return (
    <StyledPadding className={`padding_${props.multiplier}`}>
      {props.children}
    </StyledPadding>
  );
}

export default (props: { hero?: boolean; children: any }) => {
  let Area = props.hero ? HeroArea : StandardArea;

  return <Area>{props.children}</Area>;
};
