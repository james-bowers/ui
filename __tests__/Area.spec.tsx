import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

import Area, { Padding } from "../components/Area";

describe("<Area />", () => {
  test("renders Area", () => {
    const tree = shallow(<Area>example Area</Area>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  test("renders Padding", () => {
    const tree = shallow(<Padding multiplier={3}>example Area</Padding>);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
