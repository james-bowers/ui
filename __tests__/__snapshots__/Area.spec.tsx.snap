// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Area /> renders Area 1`] = `
<styled.div>
  example Area
</styled.div>
`;

exports[`<Area /> renders Padding 1`] = `
<styled.div
  className="padding_3"
>
  example Area
</styled.div>
`;
